_pages/docs/testing-with-xquery.md

```diff
@@ -5,17 +5,233 @@ lead_text: ''
 permalink: /docs/testing-with-xquery/
 ---
 
-# Testing With XQuery
+## Testing XQuery With XQuery
 
-*TODO*
+### Scenario 1 - Test assert all exist with assert-equal for success
+```
+xquery version "1.0-ml";
 
-## Testing XQuery With XQuery
+import module namespace test="http://marklogic.com/roxy/test-helper" at "/test/test-helper.xqy";
 
-*TODO*
+let $count := 2
+let $testSequence := (1,2)
+let $result := test:assert-all-exist($count, $testSequence)
+return test:assert-equal($result/@type/string(), "success")
```

Comment:
I don't think we want to assert on the result, instead we should just return the `assert-all-exist` call.  I also think it's a good practice to write failure messages for each assertion, we can train that behavior by adding them to our examples.

```
let $count := 2
let $testSequence := (1,2)
return test:assert-all-exist($count, $testSequence, "Did not find expected number of items in the sequence")
```